### DISCUSSION TITLE
Do we need to preserve gradle configuration while zipping?

### DISCUSSION THREAD

**Discussion by DelusionArun** [CastleLab/COMP3021-2021Fall-PA1-Student-Version#33] [Q&A] [answered]
For testing purposes, I unzipped the zip file I submitted for project and tried opening it with intelliJ.
Although I could still do test cases, the file no longer had the gradle configurations it did before (choice to run with CodeStyle, JAR, Java 16.0.2 SDK/JDK etc.). 
Is it normal? 
Main seems to no longer work in this version because of missing settings. Although JAR works just fine.

**Comment by Derppening** [comment, marked as answer]
You do not need to preserve the IntelliJ Run Configurations while zipping. They are provided for you merely for convenience.

In case you would like to know why, this is likely because when you zip the project, you did not zip the `.idea` files, which is where the configurations reside in.

*Note: When I grade your submission, I will just call Gradle directly to get the grading result (i.e. by `./gradlew test`). The IntelliJ Run Configurations serves as an abstraction so that you do not need to know which tasks in Gradle corresponds to which functionality.*
